### DISCUSSION TITLE
Reducing client bundle size

### DISCUSSION THREAD

**Discussion by jamauro** [meteor/meteor#13229] [Feature Requests]
Meteor is great but it also has a pretty large client-side footprint by default. I'm wondering what optimizations can be made to Meteor's core.

For example, it looks like the largest offender is `sockjs`. It seems like it was mainly put in place to have a polling fallback for browsers that didn't support websockets. Is it still needed in today's landscape? Could native websockets be used? Or if a package is needed for other reasons, could it be replaced with an alternative like `isomorphic-ws` or `uWebSockets`? Separately,  it looks like `sockjs` is still included when `DISABLE_SOCKJS=1` is set. Maybe a simple solution here would be to disable it by default, not include it in the client bundle, and if someone needs to support ancient browsers, they can enable it.

I'm sure there are other potential optimizations to be made. Curious to hear thoughts from the core team or others who have ideas here.

**Comment by jeffreyflynt** [comment]
@jamauro What frontend are you using? Blaze, React, etc?

**Reply by jamauro** [reply]
I'm using Svelte but I'm interested in what optimizations / changes could be made that are frontend-agnostic.

**Comment by xet7** [comment]
## a) Change package from clientside to serverside

https://github.com/wekan/wekan/pull/3871

##  b) Delete dependencies of package and include only required files

https://github.com/wekan/wekan/commit/23e5e1e3bd081699ce39ce5887db7e612616014d
